orchestrator/mocks/reverse_proxy_mock.go

```diff
@@ -0,0 +1,32 @@
+package mocks
+
+import (
+	"github.com/SanteonNL/orca/orchestrator/lib/coolfhir"
+	"github.com/rs/zerolog/log"
+	"net/http"
+	"net/url"
+)
+
+// MockReverseProxy is a mock implementation of a ReverseProxy for unit tests.
+type MockReverseProxy struct {
+	Proxy coolfhir.HttpProxy
+}
+
+func NewMockReverseProxy(target *url.URL, orcaPublicUrl *url.URL, transport http.RoundTripper, allowCaching bool) (*MockReverseProxy, error) {
```

Comment:
`coolfhir.HttpProxy` implements `ReverseProxy`, so this type only exists for the defaults below?

Reply:
Yeah I tried to get this to work across the unit and integ test before realising I shouldn't use it for the integ test, and I think based on how I set it up for the integ test I might be able to do similar for the unit test. Will check

Reply:
Yeah this was leftovers from when it was genuinely a mock client and not just a wrapper, I have removed it